csharp/ql/src/Dead Code/DeadStoreOfLocal.ql

```diff
@@ -72,7 +72,12 @@ predicate mayEscape(LocalVariable v) {
 
 class RelevantDefinition extends AssignableDefinition {
   RelevantDefinition() {
-    this instanceof AssignableDefinitions::AssignmentDefinition
+    this.(AssignableDefinitions::AssignmentDefinition).getAssignment() =
+      any(Assignment a |
+        if a = any(UsingStmt us).getAVariableDeclExpr()
+        then not a.getTargetVariable().hasName("_")
+        else any()
```

Comment:
Do you think we should limit this to `_` only? `_` is not a discard here, it's simply an identifier, isn't it? I think `using` variable declarations could be ignored completely.

In case of a `using` block, you could do this:
```
using(new System.IO.FileStream("", System.IO.FileMode.Open))
{
}
```
so here, introducing a variable declaration that's not referenced later is really a dead store. But skipping the declaration with a using declaration statement is not an option.

Reply:
Good point. I'll update the query to simply ignore all using declarations.